apisix/plugins/opa.lua

```diff
@@ -94,8 +95,27 @@ function _M.access(conf, ctx)
         return 503
     end
 
-    if not data.result then
-        return 403
+    local result = data.result
+
+    if not result.allow then
+        if result.headers then
+            core.response.set_header(result.headers)
+        end
+
+        local status_code = 403
+        local reason = ""
+
+        if result.status_code then
+            status_code = result.status_code
+        end
+
+        if result.reason then
+            reason = type(result.reason) == "table"
```

Comment:
Let's add test for result.reason (string and table cases).

Reply:
added